enhancements/network/networking-console-plugin.md

```diff
@@ -0,0 +1,148 @@
+
+
+### Goals:
+
+Enhanced Development Efficiency: By isolating networking console pages into a dedicated repository, developers can expedite development cycles and foster faster iterations, ultimately enhancing overall development efficiency.
+
+Improved Maintainability: Segregating the networking console pages from the main console repository will simplify maintenance and testing efforts.
+
+Dedicated Team: Currently, backend networking features are progressing faster than the corresponding UI console pages. This change will make the CNV UI team directly responsible for enhancing pages and making future features easy to use and discover.
+
+### Non-Goals
+
+N/A
+
+
+## Proposal
+
+Proposal:
+
+To execute the migration of networking console pages to the "networking-console-plugin" repository and integrate them into the cluster-networking-operator, we propose the following steps:
+
+Create "networking-console-plugin" Repository (Done): Establish a new repository named "networking-console-plugin" to house the codebase for networking console pages. This repository will serve as the centralized location for all networking-related functionalities.
+
+Move Networking Console Pages (Done): Transfer relevant code files and resources from the console repository to the "networking-console-plugin" repository and refactor them. Ensure accurate migration of dependencies and configurations to maintain seamless functionality.
+
+Integration with Cluster-Networking-Operator: Modify the cluster-networking-operator to incorporate the networking console plugin as an operand. Update configurations and references accordingly to seamlessly integrate the networking pages with the operator.
+
+Incorporation into OpenShift Release Payload: Upon successful migration and integration, include the networking console plugin in the OpenShift release payload. This ensures the plugin is delivered by default with the console, providing users immediate access to enhanced networking functionalities. Cluster-networking-operator will enable the plugin by default. The Cluster-monitor-operator is doing the same thing with its plugin [here](https://github.com/openshift/cluster-monitoring-operator/blob/master/pkg/client/client.go#L1710). Until we have "Red Hat signed" plugins there is no other solution at the moment.
+
+
+
+### Workflow Description
+
+N/A
+
+### API Extensions
+N/A
+
+### Topology Considerations
+
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+This means extra work in the beginning to integrate the operatand with the operator. 
+
+## Test Plan
+
+E2e tests will run at every pr on the networking-console-plugin using prow just like other console dynamic plugins
+
+## Graduation Criteria
+
+N/A
+
+
+### Dev Preview -> Tech Preview
+
+N/A
```

Comment:
Is this dev preview aimed behind a feature gate?

Reply:
No there is no feature gate

Reply:
we cannot merge features without feature gate in ocp.. please plan and add it in your enhancement

Reply:
We as UI team do not add ocp feautres. We just create UI to use them. We are moving part of the code and refactoring it. For the first version, the user will not even notice the difference. 
The features that I'm mentioning in this document are 'UI features' like 'clicking this button you'll create a Route'.

Reply:
Routes are part of ingress operator, so shouldn't that piece go there? cc @Miciah

Reply:
I don't want to split this section into such small pieces. It's not the best choice to move faster in the future. Creating a plugin repo with it's configuration and so on, is not that simple and to not justify pieces so small. We already talk with the SVN team and they agreed to have this into their operator...

Reply:
the plugin can even hide some pages if some functionalities are not installed in the cluster.

For example if ingress operator is not installed we can hide routes if you think is the best solution. 
But keep in mind that right now the user can see those pages either way